README.md

```diff
@@ -109,23 +109,24 @@ A valid transport (one that follows the interface defined) must implement the fo
 
 ### Creating a transport instance
 
-- `JavaScript` - `const transport = new Transport([options])`
+- `JavaScript` - `const transport = new Transport({ upgrader, ...[options] })`
 
-Creates a new Transport instance. `options` is an optional JavaScript object that should include the necessary parameters for the transport instance.
+Creates a new Transport instance. `options` is an JavaScript object that should include the necessary parameters for the transport instance. Options **MUST** include an Upgrader instance, as Transports will use this to return `interface-connection`
+instances from `transport.dial` and the listener `handlerFunction`.
 
 **Note: Why is it important to instantiate a transport -** Some transports have state that can be shared between the dialing and listening parts. For example with libp2p-webrtc-star, in order to dial a peer, the peer must be part of some signaling network that is shared with the listener.
 
 ### Dial to another peer
 
-- `JavaScript` - `const conn = await transport.dial(multiaddr, [options])`
+- `JavaScript` - `const connection = await transport.dial(multiaddr, [options])`
 
 This method uses a transport to dial a Peer listening on `multiaddr`.
 
 `multiaddr` must be of the type [`multiaddr`](https://www.npmjs.com/multiaddr).
 
 `[options]` the options that may be passed to the dial. Must support the `signal` option (see below)
 
-`conn` must implement the [interface-connection](https://github.com/libp2p/interface-connection) interface.
+Dial **MUST** call and return `upgrader.upgradeOutbound({ source, sink, conn, remoteAddr })`. The upgrader will return an [interface-connection](https://github.com/libp2p/interface-connection) instance.
```

Comment:
Do you think it's worth creating a definition of `{ source, sink, conn, remoteAddr }` as `MultiaddrConnection` somewhere so we can refer to it better here?

Reply:
Yes, I was thinking interface-connection initially, but I think it's better in the readme here since it's specific to transports.